### ISSUE TITLE
Display charts on library

### ISSUE THREAD

**Issue by danicatalan** [huridocs/uwazi#1110]
- [x] Add button to toggle between Library list and charts.
- [x] When charts tab is activated, display one graph with the first property activated (in the example below, "Super powers").
- [x] Show a chart for each property available on the sidepanel (select and multiselects).
- [x] Allow users to change between bars and pies.
- [x] Sort the results from higher to lower (as in filters)
- [x] Optional to research: add a function to detect chart size and maybe cluster values.
- [ ] Optional: Allow users change the property to render on the graph.

![screen shot 2017-07-11 at 09 39 51](https://user-images.githubusercontent.com/6364153/28056335-f4fa2dea-661c-11e7-97c5-bc79fb2ec456.png)

**Comment by RafaPolit** [comment]
Toggle between types is working nicely, as is the clustering of fields.  I reintroduced the legend for the pie now that clustering prevents dozens of results, as well as the center value on the chart using ellipsis truncated to 14 characters.

For most results, this is nice and clear.  For very long texts, the legend and the custom side tooltip are enough.

There is one additional thing:
- [ ] Test everything!!!!

Components made during the chart library decision are not yet tested. This will take a while.

This is the current look:
![screenshot from 2017-07-11 16-35-04](https://user-images.githubusercontent.com/4309019/28091943-4dfa73b6-6657-11e7-8572-5cc89ab9ebd9.png)

**Comment by danicatalan** [comment]
Yeah, the legend works well now, thanks. I think we could move to one side in the future because the readability is nicer and the space is better used. Anyway not to implement in this issue because that change will consume a lot of time fixing bugs on responsive + long names on legend and tooltip.

![image](https://user-images.githubusercontent.com/6364153/28110891-0463fdb0-66f4-11e7-9884-481ec5df98f2.png)

We should remove the text inside the pie. Doesn't make sense with the tooltip and is just adding noise to the visualisation.

Lastly, in my computer the mouse hover is not working properly. The tooltip only appears when clicking on the pie. Mouse hover is working properly on bar charts instead. @RafaPolit is it working for you?

**Comment by RafaPolit** [comment]
Yes, this is a fix they introduced with alpha.2, you should yarn install to see the mouse over functioning correctly, I updated the package.json yesterday.

I think the legend on the right might work now that we have limited the number of pie slices with the clustering, but agree that this requires some thinking regarding responsiveness.

Lastly, I think that keeping the central label is a good idea precisely because on some mobile sizes the text of the labels gets hard to read, this could be a middle-ground solution.

But we don't have to decide all that up front, we could change things as we evolve the charts and get more familiar with the library.

**Comment by txau** [comment]
Can we have colors and legends in the bar graph as well?

**Comment by RafaPolit** [comment]
Not in any library I found, as multi-color bars are reserved for multi-column or stack-column bar charts.

So, a single thread of data has a particular color.  If you were to compare two different sets of data (two columns, for instance Income and Expenses), for every month, then you would get two colors of bars, repeated for every month.

That said, we could implement it, as I agree that it would be more consistent with the Pie look.  I did try to do a quick demo of that, but, as is the case with this library, it required custom components.  It is the strength of the library, but it is also requires more time.

I'll look further into it, but it was not as straightforward as I would have liked for this particular scenario.

**Comment by txau** [comment]
@RafaPolit ok, don't invest too much into it. How about the legends? having to hover is not so usable.

**Comment by RafaPolit** [comment]
Yes.  The problem with legends and no colors is that every legend renders the same color!

But the library is versatile.  I'm glad we went this route, as the options to extend it are really cleverly thought.

Here's the result with legend and colors.  Still, legend interaction is not implemented, but I guess it shouldn't be all that hard:

![screenshot from 2017-07-14 12-13-08](https://user-images.githubusercontent.com/4309019/28222666-1c613716-688e-11e7-9477-396969896f11.png)

**Comment by danicatalan** [comment]
@RafaPolit do you need more colours for every option inside this charts?

**Comment by RafaPolit** [comment]
@dani, I have added two more, which brings the total to 8 colors.  Perhaps yes, we need a broader palette? Perhaps, just as the template types, we could alternate them for better readability (although the ´rainbow´ look is really nice, its not necessarily the most readable option)

**Comment by danicatalan** [comment]
I made a few tests, the first one is with an expanded palette using our current document/entity type colours. I really think it doesn't add so much value to our charts, so we can keep the current colours you have there. What I would like to do is review all our palette in the overall app, but this need to be achieved after implementing this.

![currentpalete](https://user-images.githubusercontent.com/6364153/28358029-a3dfeea2-6c6d-11e7-882d-7919b837e5ef.png)

What I really like is the idea of breaking the rainbow. It looks nice, but also with altered colours (I just change the order of them to be as opposite as possible). What do you think?

![altcolors](https://user-images.githubusercontent.com/6364153/28357980-77c81c9a-6c6d-11e7-97df-585df67b890b.png)

**Comment by RafaPolit** [comment]
@danicatalan Actually, in 'rainbow' mode, I think it not only looks better, but it is clearer where the pattern starts again, so you can better track in which repetition you are in.

![screenshot from 2017-07-20 10-29-57](https://user-images.githubusercontent.com/4309019/28425601-66f66272-6d36-11e7-84ed-55ade28fcef3.png)
![screenshot from 2017-07-20 10-29-58](https://user-images.githubusercontent.com/4309019/28425604-69bbb82c-6d36-11e7-9fd4-bc43e2637f60.png)

So I am keeping the rainbow for the time being, we can always change it later.

**Comment by danicatalan** [comment]
Do you think that repeat pattern is useful for the users?

I'm agree that looks better on the bar, but with a few properties (that pie, for example) the second option works great.

**Comment by kjantin** [comment]
Reopening this issue so we can continue to review internally. The features are being deployed, but we'll hide the buttons for now.

**Comment by kjantin** [comment]
Currently, if a user clicks on a property that doesn't have additional properties, the chart/graph doesn't provide any information about that selection. For example, if a user clicks on the entity type "country" for the list of main filters, the chart/graph still shows the main filters because the country entity is simply a list of 30 different countries. 

![chart-country](http://g.recordit.co/Bd0mPcomBg.gif)

I think the solution is to figure out how to take the "country" entity (and similar entities) out of that main, default filter. The only reason anyone would want to filter that is to see a list of countries - and that's the only info you would get. I think we need to change our approach to those default filters. Or, allow the admin to take those kinds of entities out of the main filter (just like we do for other properties, we ask if they want this property to show up in filters).

**Comment by emamut** [comment]
IMHO I think that the percentages should be in the tags
![image pasted at 2017-7-25 09-39](https://user-images.githubusercontent.com/134828/28690305-4f2cd4d4-72de-11e7-8810-619389b875af.png)

**Comment by simonfossom** [comment]
@txau is this issue still relevant?

**Comment by txau** [comment]
@simonfossom yes its still relevant. We want to develop this at some point, although I think we should run a UX research before.
